### ISSUE TITLE
static fms not declaring OpenMP library dependencies correctly

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#923]
**Describe the bug**
When fms is used as a static library, macOS users with apple-clang always have to explicitly add the linking to the OpenMP library:
```
cmake -DCMAKE_SHARED_LINKER_FLAGS="${llvm_openmp_ROOT}/lib/libomp.dylib"
```
I think that the static fms library doesn't declare it's dependency on the LLVM OpenMP library correctly.

**To Reproduce**
Compile ufs-weather-model or ufs-bundle on macOS with apple-clang and spack-stack, and watch for the final linker error:
```
Undefined symbols for architecture arm64:
  "___kmpc_for_static_fini", referenced from:
      _.omp_outlined..1 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..3 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..6 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..7 in libfms_r8.a(create_xgrid.c.o)
  "___kmpc_for_static_init_4", referenced from:
      _.omp_outlined..1 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..3 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..6 in libfms_r8.a(create_xgrid.c.o)
      _.omp_outlined..7 in libfms_r8.a(create_xgrid.c.o)
  "___kmpc_fork_call", referenced from:
      _create_xgrid_2dx2d_order1 in libfms_r8.a(create_xgrid.c.o)
      _create_xgrid_2dx2d_order2 in libfms_r8.a(create_xgrid.c.o)
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status
```

**Expected behavior**
FMS tells the `cmake` build system that it needs to link against the LLVM OpenMP library on macOS with apple-clang.

**System:**
macOS

**Additional context**
n/a

**Comment by climbfuji** [comment]
This FMS PR fixes the problem on macOS: https://github.com/NOAA-GFDL/FMS/pull/1434

**Comment by climbfuji** [comment]
This can be closed, because the update to fms 2024.01 #1106 will take care of it.
